nixos/modules/services/databases/ghidra-server.nix

```
@@ -0,0 +1,264 @@
+        default = [
+          "TLS_DHE_RSA_WITH_AES_256_GCM_SHA384"
+          "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384"
+          "TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384"
+          "TLS_AES_256_GCM_SHA384"
+        ];
+        description = "";
+      };
+    };
+
+    authentication = {
+      type = lib.mkOption {
+        type = lib.types.enum [
+          "local"
+          "pki"
+          "jaas"
+        ];
+        default = "local";
+        description = "";
+      };
+
+      useClientLogin = lib.mkEnableOption "the use of the client-side login ID";
+
+      sshKey = lib.mkEnableOption "the use of SSH pre-shared key for authentication when the local login method is used";
+
+      useAutoProvision = lib.mkEnableOption "the auto-creation of new users when they successfully authenticate to the server. Only works with login methods local and jaas";
+
+      allowAnonymous = lib.mkEnableOption "anonymous access support for Ghidra Server and its repositories";
+    };
+
+    console = {
+      title = lib.mkOption {
+        type = lib.types.str;
+        default = "Ghidra Server";
+        description = "";
+      };
+
+      format = lib.mkOption {
+        type = lib.types.str;
+        default = "PM";
+        description = "";
+      };
+
+      loglevel = lib.mkOption {
+        type = lib.types.enum [
+          "NONE"
+          "FATAL"
+          "ERROR"
+          "STATUS"
+          "INFO"
+          "DEBUG"
+        ];
+        default = "INFO";
+        description = "";
+      };
+    };
+
+    logfile = {
+      enable = lib.mkEnableOption "logging to a file";
+
+      loglevel = lib.mkOption {
+        type = lib.types.enum [
+          "NONE"
+          "FATAL"
+          "ERROR"
+          "STATUS"
+          "INFO"
+          "DEBUG"
+        ];
+        default = "INFO";
+        description = "";
+      };
+
+      format = lib.mkOption {
+        type = lib.types.str;
+        default = "PM";
+        description = "";
+      };
+
+      maxFiles = lib.mkOption {
+        type = lib.types.int;
+        default = 10;
+        description = "";
+      };
+
+      maxSize = lib.mkOption {
+        type = lib.types.str;
+        default = "10m";
+        description = "";
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    users.groups."${cfg.group}" = { };
+    users.users."${cfg.user}" = {
+      isSystemUser = true;
+      home = cfg.repositoryDir;
+      shell = "/run/current-system/sw/bin/sh";
+      group = cfg.group;
+      packages = [
+        cfg.javaPackage
+      ];
+    };
+
+    systemd.tmpfiles.rules = [
+      "L+ /etc/ghidra-server.conf - - - - ${ghidraServerConfig}"
+    ];
+
+    systemd.services.ghidra-server = {
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      path = [ cfg.javaPackage ];
+      serviceConfig = {
+        Type = "simple";
+        User = cfg.user;
+        Group = cfg.group;
+        TimeoutSec = "300";
```

With the service marked as `Type=simple`, I believe it will be considered started from the moment `ExecStart` has been run.

```suggestion
        TimeoutStopSec = "300";
```